### DISCUSSION TITLE
`pip install -e .[dev,doc,test]` in contributing documention wont work, is this a documentation error?

### DISCUSSION THREAD

**Discussion by csrgxtu** [fastapi/fastapi#6117] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
pip install -e .[dev,doc,test]
```

### Description

![image](https://user-images.githubusercontent.com/5053620/191152983-0b0161b0-0562-4120-8931-4e0993427b5c.png)

i am trying to execute upper command in my local python virtual env, and got error return:

![image](https://user-images.githubusercontent.com/5053620/191153106-2e2c09b2-3ce4-470e-9918-77840fcb47ad.png)

### Operating System

Linux

### Operating System Details

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04.1 LTS"

### FastAPI Version

0.85.0

### Python Version

3.10.7

### Additional Context

i cloned latest fastapi, trying to understand how to generate docs, when i follow the following documention:
https://fastapi.tiangolo.com/contributing/

esp when running following cmd:
```bash
pip install -e .[dev,doc,test]
```
got following result
```bash
zsh: no matches found: .[dev,doc,test]
```

what should i do? is this a documentation error?

**Comment by csrgxtu** [comment]
`pip install -e` will install current file into local virtualenv, but what dose `.[dev,doc,test]` mean?

**Comment by csrgxtu** [comment]
![image](https://user-images.githubusercontent.com/5053620/191159241-d4b5ac6f-c50f-4748-8c97-325e7b159c46.png)

**Comment by csrgxtu** [comment]
`pip install -e .[dev,doc,test]` should be `pip install -e fastapi` fastapi is the repo you just cloned from github and this cloned fastapi will be installed into current virtual env, and also when you edit fastapi source files, it can be used immediately.
![image](https://user-images.githubusercontent.com/5053620/191159378-407042ea-bf4d-4185-ab3c-de172483f831.png)

**Comment by csrgxtu** [comment]
to successfully start local documentation server, following requirements need to be installed:
```text
mkdocs
typer
pymdown-extensions
mkdocs-material
mdx_include
mkdocs-markdownextradata-plugin
```

**Comment by JarroVGIT** [comment]
This [answer](https://stackoverflow.com/a/69977707/1557060)  on SO explains what `.[test,dev,docs]` does; it installs the optional dependencies in `pyproject.toml`.

It’s not a documentation error, FastAPI recently moved away from Flit, and this change was part of it. Can’t test it right now as I am on my phone, but maybe you can try it in bash rather than zsh?

**Comment by JarroVGIT** [comment]
I just tested here locally, it doesn't seem to work on `zsh` but it does work on `bash`. As the docs also say to use bash, I don't think it's an error in the docs.

For using zsh, try this command: `pip install -e '.[dev,docs,test]'`. The single quotes will ensure that zsh will not try to use the bit between the brackets for pattern matching but takes the brackets as literal brackets.

**Comment by csrgxtu** [comment]
@JarroVGIT thanks, in `Bash` it works as documentation said.
![image](https://user-images.githubusercontent.com/5053620/191184454-25c5b827-3d73-4f02-a964-fd84daa231ea.png)

**Comment by csrgxtu** [comment]
> Flit is a simple way to put Python packages and modules on PyPI.

https://flit.pypa.io/en/stable/

**Comment by csrgxtu** [comment]
> I just tested here locally, it doesn't seem to work on `zsh` but it does work on `bash`. As the docs also say to use bash, I don't think it's an error in the docs.
> 
> For using zsh, try this command: `pip install -e '.[dev,docs,test]'`. The single quotes will ensure that zsh will not try to use the bit between the brackets for pattern matching but takes the brackets as literal brackets.

@JarroVGIT should  `pip install -e .[dev,doc,test]` replaced by `pip install -e '.[dev,doc,test]'` in documentation to support all shell as default?

**Comment by iudeen** [comment, marked as answer]
FastAPI does not use Flit anymore.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue 👍
